src/mrpro/operators/_FourierOp.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import torch
+import torch.nn.functional as F
+from einops import rearrange
+from einops import repeat
+from torchkbnufft import KbNufft
+from torchkbnufft import KbNufftAdjoint
+
+from mrpro.data import KTrajectory
+from mrpro.data import SpatialDimension
+from mrpro.operators import LinearOperator
+
+
+def is_uniform(x: torch.Tensor, atol=1e-6):
+    """Check if a tensor contains equidistant sampling points.
+
+    If the tensor has batch-dimension>1, this is checked for all
+    batches.
+    """
+    # TODO: currrently, this function is never used here
+
+    all_close_list = [
+        torch.allclose(torch.diff(x[kb, ...]), torch.diff(x[kb, ...])[0], atol=atol) for kb in range(x.shape[0])
+    ]
+    return torch.tensor(all_close_list).all()
+
+
+class FourierOp(LinearOperator):
+    """Fourier operator class."""
+
+    def __init__(
+        self,
+        im_shape: SpatialDimension[int],
+        traj: KTrajectory,
+        oversampling: SpatialDimension[float] = SpatialDimension(
+            2.0, 2.0, 2.0
+        ),  # TODO: not the nicest, since only used for nuFFT
+        norm='ortho',
```

Comment:
> I think it might be the easiest to require the k-space data to be acquired by a Fourier-transform that has norm="ortho" and if not, to appropriately weight it before using the operator

Yes, I think so too

Reply:
ok, I'll remove it then for the moment

Reply:
...and include `kbwidth` and `numpoints` as arguments